src/revocation.rs

```diff
@@ -0,0 +1,477 @@
+use crate::did_resolve::DIDResolver;
+use crate::jsonld::REVOCATION_LIST_2020_V1_CONTEXT;
+use crate::one_or_many::OneOrMany;
+use crate::vc::{Credential, CredentialStatus, Issuer, VerificationResult, URI};
+use async_trait::async_trait;
+use bitvec::prelude::Lsb0;
+use bitvec::slice::BitSlice;
+use bitvec::vec::BitVec;
+use core::convert::TryFrom;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use thiserror::Error;
+
+type URL = String;
```

Comment:
Because `reqwest` is a non-optional dependency you can use `reqwest::Ur`

Reply:
Although I guess the validation is already covered with the json-ld deserialisation?

Reply:
This `URL` type is used in one place, that is the `revocationListCredential` (`revocation_list_credential`) property of a [RevocationList2020Status](https://w3c-ccg.github.io/vc-status-rl-2020/#revocationlist2020status) [credential status](https://www.w3.org/TR/vc-data-model/#status) object.
Doing more checking of the URL would be good. A `request::Url` instance could be passed directly to `request::get`. But `request::Url` doesn't implement `Serialize` or `Deserialize`, only `Display` and `FromStr`; I would probably wrap it in a struct with `try_from` and `into` serde attributes, as done elsewhere in this repo, unless there is a better way.

This property `revocationListCredential`, and the properties in `RevocationList2020Status`, were not validated in JSON-LD. I've added a fix for this in eb7218412099a378ed573b457b56ed76423913d0 - to require that credentials using RevocationList2020 use the specific context URI for RevocationList2020.

Reply:
I think you can implement deserialisation for a type without having to wrap it -- but if it's only used in one place for now, and we actually check its value I'd say it's fine as it is for now :+1: